### PR TITLE
[SCSB-145] require Python 3.10

### PR DESCRIPTION
**Pull request by zacharyburnett**
resolves [SCSB-145](https://jira.stsci.edu/browse/SCSB-145)

propagate Astropy's deprecation of Python 3.9 to downstream packages

> [!NOTE]
> this PR was generated automatically by ``batchpr`` :robot:

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -93,80 +101,76 @@ src = [
...
+    "ARG",
+    "ERA",
+    "PGH",
+    "FLY",
+    "NPY",
+    "PERF",
+    "RUF",
 ]
 ignore = [
-    "ISC001",  # conflicts with ruff formatter
```
Comment (tapastro) [inline pyproject.toml:137]: Ditto

**Review by tapastro — CHANGES_REQUESTED** [review]
I think we might as well try to get this merged before we make a release for jwst b10.2. Just some changes requested on lint/black/? removing useful comments.

**Comment by zacharyburnett** [comment]
> I think we might as well try to get this merged before we make a release for jwst b10.2. Just some changes requested on lint/black/? removing useful comments.

no problem, fixed in 08eb649d86f3e148138d96e34509d79895659ae1; I'll fix the script so it preserves comments in the future

**Review by tapastro — APPROVED** [review]
Looks good to me!

**Pull request merged**